handler/endpoint.go

```diff
@@ -102,6 +102,14 @@ func (e *Endpoint) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	e.readResults(subCtx, proxyResults, beresps)
 	e.readResults(subCtx, requestResults, beresps)
 
+	select {
```

Comment:
why select here?

Reply:
to check a possible cancel before going further